cloudinit/tests/test_features.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+# pylint: disable=no-member,no-name-in-module
+"""
+This file is for testing the feature flag functionality itself,
+NOT for testing any individual feature flag
+"""
+import pytest
+from importlib import reload
+from pathlib import Path
+
+import cloudinit
+import cloudinit.features
+
+
+@pytest.yield_fixture()
+def create_override(request):
+    override_path = Path(cloudinit.__file__).parent / 'feature_overrides.py'
+    with override_path.open('w') as f:
+        for key, value in request.param.items():
+            f.write('{} = {}\n'.format(key, value))
+
+    # The goal is to reload features.py so we also get the
+    # features_overrides.py stuff. This could normally be achieved with
+    # a single reload(cloudinit.features), however, since this fixture will be
+    # called multiple times in the same interpreter, calling
+    # reload(cloudinit.features) a second time won't refresh what already
+    # exists in feature_overrides.py because that's now already defined, so we
+    # need to call reload(cloudinit.feature_overrides) first. But the first
+    # time we call this fixture, cloudinit.feature_overrides doesn't exist yet,
+    # so we need to reload cloudinit.features first before we can reload
+    # cloudinit.feature_overrides, hence the three calls here.
+    reload(cloudinit.features)
+    reload(cloudinit.feature_overrides)
+    reload(cloudinit.features)
```

Comment:
I think we can actually simplify this a little by removing the internally-cached `feature_overrides` module if present, then reloading `cloudinit.features` (which will then have to reimport our updated `feature_overrides`:

```suggestion
    sys.modules.pop('cloudinit.feature_overrides', None)
    reload(cloudinit.features)
```

and we could potentially even do the same for `cloudinit.features` (but that could cause problems with other tests so I wouldn't recommend it).